### PR TITLE
Refactor polaris-stack

### PR DESCRIPTION
**Pull request by vladucu**
### What's here

- refactors to ES6 class
- template-only components
- angle-bracket syntax

NOTE: Spent a bit of time looking for a solution to our auto-wrapping thingy and ultimately landed on using `MutationObserver`s. This seems to work well.

However, in the long term I'm wondering if we should think about just having a codemod and make usage explicit everywhere? Will be much more verbose for sure.....so not sold on it
Curious what you're thoughts are

### CODE REVIEW

File: addon/templates/components/polaris-drop-zone/file-upload.hbs [hunk 1]
```diff
@@ -15,9 +15,7 @@
     <PolarisStack @vertical={{true}} @spacing="tight">
       {{#if (eq this.type "file")}}
         <img class={{this.imageClasses}} src={{this.fileUpload}} alt="">
-      {{/if}}
-
-      {{#if (eq this.type "image")}}
+      {{else if (eq this.type "image")}}
```
Comment (andrewpye) [inline addon/templates/components/polaris-drop-zone/file-upload.hbs]: lol, not quite sure how these ended up falling under the `polaris-stack` refactor 🙈😛

File: addon/templates/components/polaris-stack.hbs [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+  {{did-insert this.setupAutoWrapper}}
+  {{will-destroy this.teardownAutoWrapper}}
+  ...attributes
+>
+  {{#if hasBlock}}
+    {{yield (hash item=(component "polaris-stack/item"))}}
+  {{else}}
+    {{text}}
+  {{/if}}
+</div>
```
Comment (andrewpye) [inline addon/templates/components/polaris-stack.hbs]: Nit: missing newline

File: addon/templates/components/polaris-stack/item.hbs [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
+  data-test-stack-item={{true}}
+  class={{this.classes}}
+  ...attributes
+>
+  {{#if hasBlock}}
+    {{yield}}
+  {{else}}
+    {{text}}
+  {{/if}}
+</div>
```
Comment (andrewpye) [inline addon/templates/components/polaris-stack/item.hbs]: Nit: missing newline

**Comment by vladucu** [comment]
>  Not sure where I stand on this, which means that for now I'm good with leaving this as is if it still works 

👍

**Pull request merged**